# HG changeset patch
# User Torsten <mttl@mailbox.org>
# Date 1542310812 -3600
#      Thu Nov 15 20:40:12 2018 +0100
# Node ID 13d11afc8122e1a9c8fb6844807a1c1c01a12c01
# Parent  5464e9c1da6aff7aa8593a59ab8b6d85944702df
fix crash after editor dialog about creation of non-existing file

* file-editor.cc (request_open_file): assign nullptrto fileEditorTab after
  its deletion for later testing for valid editor tab before trying to
  restore breakpoints

diff --git a/libgui/src/m-editor/file-editor.cc b/libgui/src/m-editor/file-editor.cc
--- a/libgui/src/m-editor/file-editor.cc
+++ b/libgui/src/m-editor/file-editor.cc
@@ -1354,16 +1354,17 @@ namespace octave
                         if (breakpoint_marker)
                           emit fetab_do_breakpoint_marker (insert, fileEditorTab,
                                                            line, cond);
                       }
                   }
                 else
                   {
                     delete fileEditorTab;
+                    fileEditorTab = nullptr;
 
                     if (QFile::exists (openFileName))
                       {
                         // File not readable:
                         // create a NonModal message about error.
                         QMessageBox *msgBox
                           = new QMessageBox (QMessageBox::Critical,
                                              tr ("Octave Editor"),
@@ -1427,17 +1428,18 @@ namespace octave
                       }
                   }
               }
 
             if (! ((breakpoint_marker || debug_pointer) && is_editor_console_tabbed ()))
               {
                 // update breakpoint pointers, really show editor
                 // and the current editor tab
-                fileEditorTab->update_breakpoints ();
+                if (fileEditorTab)
+                  fileEditorTab->update_breakpoints ();
                 focus ();
                 emit file_loaded_signal ();
               }
           }
       }
   }
 
   void file_editor::request_preferences (bool)
